java/client/src/main/java/com/vesoft/nebula/client/graph/storage/general/GeneralStorageClient.java

```diff
@@ -0,0 +1,659 @@
+
+            if (!requests.containsKey(leader)) {
+                KVPutRequest request = new KVPutRequest();
+                request.setSpace_id(spaceId);
+                Map<Integer, List<KeyValue>> parts = Maps.newHashMap();
+                parts.put(part, partKeyValue.getValue());
+                request.setParts(parts);
+                requests.put(leader, request);
+            } else {
+                KVPutRequest request = requests.get(leader);
+                if (request.getParts().containsKey(part)) {
+                    request.getParts().get(part).addAll(partKeyValue.getValue());
+                } else {
+                    request.getParts().put(part, partKeyValue.getValue());
+                }
+            }
+        }
+
+        final CountDownLatch countDownLatch = new CountDownLatch(partKeyValues.size());
+        final List<Boolean> responses = Collections.synchronizedList(
+                new ArrayList<>(partKeyValues.size()));
+
+        for (final Map.Entry<HostAndPort, KVPutRequest> entry : requests.entrySet()) {
+            threadPool.submit(() -> {
+                if (doPut(spaceName, entry.getKey(), entry.getValue())) {
+                    responses.add(true);
+                } else {
+                    responses.add(false);
+                }
+                countDownLatch.countDown();
+            });
+        }
+
+        try {
+            countDownLatch.await();
+        } catch (InterruptedException interruptedE) {
+            LOGGER.error("Put interrupted:", interruptedE);
+            return false;
+        }
+
+        for (Boolean resp : responses) {
+            if (!resp) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+
+    /**
+     * wrap request and execute put
+     *
+     * @param spaceName nebula graph space
+     * @param leader    leader for part
+     * @param request   PutRequest
+     * @return boolean
+     */
+    private boolean doPut(String spaceName, HostAndPort leader, KVPutRequest request) {
+        // there is at least one retry for leader change.
+        int retry = 0;
+        while (retry++ != metaInfo.getConnectionRetry()) {
+
+            try {
+                connection = getConnection(leader);
+                if (connection == null) {
+                    LOGGER.error("connection is in usage");
+                    return false;
+                }
+                connection.setBusy();
+            } catch (Exception e) {
+                LOGGER.error("Connect error: ", e);
+                return false;
+            }
+
+            try {
+                ExecResponse result = connection.put(request);
+                if (isSuccessfully(result)) {
+                    connection.release();
+                    return true;
+                }
+                for (PartitionResult partResult : result.getResult().getFailed_parts()) {
+                    if (partResult.code == ErrorCode.E_LEADER_CHANGED) {
```

Comment:
if the error code is not leader_change, directly return, no need to retry. and better to return the error msg to users